### PR TITLE
[GUI][SETUP] Added a password step to setup wizard

### PR DESCRIPTION
**Pull request by mxaddict**
Capture2

### CODE REVIEW

**Comment by mxaddict** [comment]
related to #697

**Comment by mxaddict** [comment]
NOTES: Password step should come always at the last step before loading wallet

**Comment by navbuilder** [comment]
A new build of 9e665a836c0fa817acd3942d9f0fd822151c742d has completed succesfully!
Binaries available at https://build.nav.community/binaries/add-password-option-to-wizard

**Comment by chasingkirkjufell** [comment]
The button is "skip/next" when creating a new wallet but only "next" when importing with mnemonics. 
image

**Comment by Xa5r** [comment]
Capture
I would change it to "Yes" instead of "Ok"

**Comment by Xa5r** [comment]
> 
> 
> The button is "skip/next" when creating a new wallet but only "next" when importing with mnemonics.
> image

I think on the step on the picture, there dont need to stay skip/next, because without password is also just going to the next step

**Comment by Xa5r** [comment]
Not sure if this is a bug here, but if you dont set a password and encrypt the wallet later, the mnemonic changes

**Comment by mxaddict** [comment]
> Not sure if this is a bug here, but if you dont set a password and encrypt the wallet later, the mnemonic changes

This won't happen once the fix for that bug is also merged into master

**Comment by mxaddict** [comment]
> > The button is "skip/next" when creating a new wallet but only "next" when importing with mnemonics.
> > image
> 
> I think on the step on the picture, there dont need to stay skip/next, because without password is also just going to the next step

Ok, so just "next"?

**Comment by Xa5r** [comment]
Yes
If don't want set a password, for you it's just the next step.
If you already wrote a password, for you it's not a skip in (even he doesn't want a password). 
Meaning the state will be used anyway with or without password

**Comment by mxaddict** [comment]
> Yes
> If you don't want set a password, for you it's just the next step.
> If you already wrote a password, for you it's not a skip in (even he doesn't want a password).
> Meaning the state will be used anyway with or without password

Makes sense 👍

**Comment by mxaddict** [comment]
I've pushed the suggested changes 👍

**Comment by navbuilder** [comment]
A new build of 1cd0efda61e6509f27cdccf02c4e71da1c6fc632 has completed succesfully!
Binaries available at https://build.nav.community/binaries/add-password-option-to-wizard

File: src/qt/startoptionsmain.cpp [hunk 1]
```diff
@@ -223,5 +235,33 @@ void StartOptionsMain::on_Next_clicked() {
...
             break;
         }
+        case PasswordPage: {
+            std::string pass = startOptionsPassword->getPass().toStdString();
+            std::string passConf = startOptionsPassword->getPassConf().toStdString();
+            if (pass.empty() && passConf.empty())
+            {
+                QMessageBox msgBox;
+                msgBox.setIcon(QMessageBox::Warning);
+                msgBox.setText(tr("Are you sure you want to skip seed confirmation?"));
```
[the remark this answers is not in the document]
Comment (mxaddict) [reply src/qt/startoptionsmain.cpp:245]: How did I miss that? :P

[the remark this answers is not in the document]
Comment (mxaddict) [reply src/qt/startoptionsmain.cpp]: I've updated the message

File: src/wallet/wallet.cpp [hunk 2]
```diff
@@ -4072,6 +4072,16 @@ bool CWallet::InitLoadWallet(const std::string& wordlist)
...
+
+                // Check if we gave a password
+                if (!password.empty()) {
+                    // Is this safe? or do I need to do something more
+                    // to the string after I've converted it to SecureString
+                    SecureString strWalletPass;
+                    strWalletPass.reserve(100);
+                    strWalletPass = password.c_str();
+                    walletInstance->EncryptWallet(strWalletPass);
+                }
```
Comment (mxaddict) [inline src/wallet/wallet.cpp:4084]: @aguycalled About this part, is this safe? will it not lead the password in memory?

**Pull request merged**